src/Megamix/Commands.cpp

```diff
@@ -0,0 +1,112 @@
+#include <3ds.h>
+#include <CTRPluginFramework.hpp>
+
+#include "Megamix.hpp"
+
+using CTRPluginFramework::OSD;
+using CTRPluginFramework::Utils;
+
+namespace Megamix{
+
+    void tickflowCommandsHookWrapper() {
+        asm(
+            "mov r0, r6\n"
+            "bl tickflowCommandsHook\n"
+            "bx r0\n"
+        );
+    }
+
+    extern "C" __attribute__((used)) 
+    int tickflowCommandsHook(CTickflow* self, u32 cmd_num, u32 arg0, u32* args){
+        switch(cmd_num){
+            case 0: // Necessary, as it doesn't execute otherwise.
+                return Region::TickflowAsyncSubLocation();
+            case InputCheck:
+                input_cmd(self, arg0, args);
+            break;
+            case VersionNumber:
+                versionCheck(self, arg0, args);
+            break;
+            case LanguageCheck:
+                languageCheck(self, arg0, args);
+            break;
+            case DisplayCondvar:
+                displayCondvar(self, arg0, args);
+            break;
+        }
+        return Region::TickflowCommandsEnd();
+    }
+
+    void input_cmd(CTickflow* self, u32 arg0, u32* args) {
+        if (arg0 > 2) {
+            self->condvar = 0;
+            return;
+        } else if (arg0 == 2) {
+            CSaveData** gSaveData = (CSaveData**)Region::GlobalSaveDataPointer();
+            arg0 = (u32)(*gSaveData)->fileData[(*gSaveData)->currentFile].playStyle;
```

Comment:
add something along the lines of this, for clarity
```suggestion
            // arg0 here is the playstyle to read inputs from - so this results in input-reading being dependent on playstyle
            arg0 = (u32)(*gSaveData)->fileData[(*gSaveData)->currentFile].playStyle;
```